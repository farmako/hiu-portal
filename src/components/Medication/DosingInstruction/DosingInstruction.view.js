import React from 'react';
import * as PropTypes from 'prop-types';
import DosageTiming from '../DosageTiming';
import DosingInstructionForPatient from '../DosingInstructionForPatient';
import DosingInstructionAsNeeded from '../DosingInstructionAsNeeded';

const DosingInstruction = (props) => {
  const { dosage } = props;
  return dosage ? (
    <div>
      <span>{dosage.text}</span>
      <br />
      <DosageTiming dosage={dosage} />
      <ul className="instruction-list-item">
        <DosingInstructionForPatient dosage={dosage} />
        <DosingInstructionAsNeeded dosage={dosage} />
      </ul>
    </div>
  ) : null;
};

DosingInstruction.propTypes = {
  dosage: PropTypes.shape({ text: PropTypes.string }),
};

DosingInstruction.defaultProps = {
  dosage: null,
};


export default DosingInstruction;
