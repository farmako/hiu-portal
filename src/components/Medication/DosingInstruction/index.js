import DosingInstruction from './DosingInstruction.view';

export default DosingInstruction;
