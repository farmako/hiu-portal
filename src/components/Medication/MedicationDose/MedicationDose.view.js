import * as PropTypes from 'prop-types';
import React from 'react';
import DosingInstruction from '../DosingInstruction';

const MedicationDose = (props) => {
  const { dosageInstructions } = props;
  if (dosageInstructions && dosageInstructions.length > 0) {
    const instructions = dosageInstructions.map((instruction) => (
      <DosingInstruction key={instruction.text} dosage={instruction} />
    ));
    return <div>{instructions}</div>;
  }
  return null;
};

MedicationDose.propTypes = {
  dosageInstructions: PropTypes.arrayOf(PropTypes.shape({ text: PropTypes.string })),
};

MedicationDose.defaultProps = {
  dosageInstructions: [],
};

export default MedicationDose;
