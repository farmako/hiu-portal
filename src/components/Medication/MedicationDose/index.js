import MedicationDose from './MedicationDose.view';

export default MedicationDose;
