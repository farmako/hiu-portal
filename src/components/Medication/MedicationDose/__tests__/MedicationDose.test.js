import React from 'react';
import {shallow} from "enzyme";
import MedicationDose from "../MedicationDose.view";
import DosingInstruction from "../../DosingInstruction";

describe('MedicationDose', () => {
    const mockData = [
        {
            sequence: 1,
            text: '22/01/2020 to 07/02/2020',
            timing: {
                event: ['22/1/2019'],
            },
        }
    ];
    const wrapper = shallow(
      <MedicationDose dosageInstructions={mockData} />
    );

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(DosingInstruction).exists()).toEqual(true);
    });

    it('should not render anything if data is undefined', () => {
        wrapper.setProps({dosageInstructions:undefined});
        expect(wrapper.find(DosingInstruction).exists()).toEqual(false);
    });
});