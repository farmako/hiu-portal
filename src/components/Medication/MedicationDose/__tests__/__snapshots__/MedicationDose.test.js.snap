// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationDose should render properly 1`] = `
<div>
  <DosingInstruction
    dosage={
      Object {
        "sequence": 1,
        "text": "22/01/2020 to 07/02/2020",
        "timing": Object {
          "event": Array [
            "22/1/2019",
          ],
        },
      }
    }
    key="22/01/2020 to 07/02/2020"
  />
</div>
`;
